### ISSUE TITLE
tf.train.init_from_checkpoint should add reshape parameter like the tf.train.Saver

### ISSUE THREAD

**Issue by Flowerfan** [tensorflow/tensorflow#20345]
Have I written custom code . Yes, with Estimator API
OS Platform and Distribution. Ubuntu 16.04
TensorFlow installed from: pip
TensorFlow version: 1.8
Bazel version: N/A
CUDA/cuDNN version: 9.0
GPU model and memory: 1080Ti with 11G
Exact command to reproduce: N/A

Sometimes pretrained weights are squeezed while the weights for training have extra dimensions. For example, the shape of V1 in checkpoint is [128], while shape of V1 in target net is [1,1,1,128].  The reshape parameter in tf.train.Saver can solve this issue. But when I use Estimator to load pretrained weights, it will be very difficult to load weights with mismatch shapes.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by bignamehyp** [comment]
You might try tf.contrib.framework.assign_from_checkpoint_fn

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
As this issue has invited community support, please remove the assignee. Otherwise, remove the `community support` label. Thank you.

**Comment by tensorflowbutler** [comment]
As this issue has invited community support, please remove the assignee. Otherwise, remove the `community support` label. Thank you.

**Comment by AVCarreiro** [comment]
@Flowerfan did you manage to find a solution for this? I'm interested in any hints, please. Thanks!

**Comment by Flowerfan** [comment]
@AVCarreiro  I used scaffold and tf.train.Saver to deal with it. First create a scaffold object for initialization, and then pass it to EstimatorSpec.
Here is my solution: 
saver = tf.train.Saver(var_list=varlist, reshape=True)
def init_fn(scaffold, session):
    saver.restore(session, model_pth)
scaffold = tf.train.Scaffold(init_fn=init_fn)

tf.estimator.EstimatorSpec(mode, loss=loss, train_op=train_op, scaffold=scaffold)
